lib/GHCup/GHC.hs

```diff
@@ -1037,6 +1042,24 @@ compileGHC targetGhc crossTarget vps bstrap hghc jobs mbuildConfig patches aargs
     pure installVer
 
  where
+  bootAndConfigure :: ( MonadReader env m
+                      , HasSettings env
+                      , HasDirs env
+                      , HasLog env
+                      , MonadIO m
+                      , MonadThrow m
+                      )
+                   => GHCupPath
+                   -> Excepts '[ProcessError, NotFoundInPATH] m ()
+  bootAndConfigure tmpUnpack = do
+    let bootFile = fromGHCupPath tmpUnpack </> "boot"
+    hasBootFile <- liftIO $ doesFileExist bootFile
+    when hasBootFile $ do
+      lift $ logDebug "Doing ghc-bootstrap"
+      python3 <- liftE $ makeAbsolute "python3"
+      lEM $ execLogged python3 ["./boot"] (Just $ fromGHCupPath tmpUnpack) "ghc-bootstrap" Nothing
+      liftE $ configureWithGhcBoot Nothing [] (Just $ fromGHCupPath tmpUnpack) "ghc-bootstrap"
```

Comment:
I'm not sure this belong outside of `getGHCVer`. The reason we run configure here (without any meaningful arguments) is because it creates the VERSION file, which we then use. So this is not the "proper" configure. The only reason we do this is to get the GHC version.

See GHCs `configure.ac`:

```
dnl Create the VERSION file, satisfying #22322.
printf "$ProjectVersion" > VERSION
```

Reply:
Ok, in that case it does belong here, as the bootstrapped ghc will contain the VERSION file.